### ISSUE TITLE
`@typescript-eslint/no-shadow` maybe shouldn't trigger on Snippets?

### ISSUE THREAD

**Issue by marekdedic** [sveltejs/eslint-plugin-svelte#908]
### Description

Hi, the rule `@typescript-eslint/no-shadow` now triggers on snippets like this:

```svelte
<ComponentWithSnippet>
  {#snippet children()}
    <AnotherComponentWithSnippet>
      {#snippet children()}
        Hello!
      {/snippet}
    </AnotherComponentWithSnippet>
  {/snippet}
</ComponentWithSnippet>
```

This results on an error on the second `children` function (`children' is already declared in the upper scope on line ...`). However, this is probably (?) Mostly a false positive - you aren't actually shadowing a meaningful function or variable and you cannot really always control Snippet names (e.g. when they come from a 3rd party library) and you want to use the name `children` if possible...

Could we change the rule/parsing/...? so that this rule never triggers on snippets? I can't come up with an example where you'd actually want the rule to trigger on snippets, but maybe there is some?

**Comment by marekdedic** [comment]
Hmm, sorry, now I found #871, which is very similar....
